### PR TITLE
[flang] Allow different linker name

### PR DESCRIPTION
**Pull request by kkwli**
The linker name is different if  the `lld` project is enabled `-DLLVM_ENABLE_PROJECTS="...;lld"`.

### CODE REVIEW

File: flang/test/Driver/isysroot.f90 [hunk 1]
```diff
@@ -6,7 +6,7 @@
 ! RUN: %flang -### --target=aarch64-linux-gnu -isysroot /path/to/sysroot \
 ! RUN:        %s 2>&1 | FileCheck %s --check-prefix=CHECK-LINUX
 
-! CHECK-DARWIN: "{{.*[\/]}}ld" {{.*}}"-syslibroot" "/path/to/sysroot"
+! CHECK-DARWIN: "{{.*[\/]}}ld{{(64)?(.lld)?}}" {{.*}}"-syslibroot" "/path/to/sysroot"
```
Comment (luporl) [inline flang/test/Driver/isysroot.f90:9]:
```suggestion
! CHECK-DARWIN: "{{.*[\/]}}ld{{(64)?(\.lld)?}}" {{.*}}"-syslibroot" "/path/to/sysroot"
```

**Review by luporl — APPROVED** [review]
LGTM, thanks for this correction.

**Review by banach-space — APPROVED** [review]
Thanks!

File: flang/test/Driver/isysroot.f90 [hunk 2]
```diff
@@ -6,7 +6,7 @@
 ! RUN: %flang -### --target=aarch64-linux-gnu -isysroot /path/to/sysroot \
 ! RUN:        %s 2>&1 | FileCheck %s --check-prefix=CHECK-LINUX
 
-! CHECK-DARWIN: "{{.*[\/]}}ld" {{.*}}"-syslibroot" "/path/to/sysroot"
+! CHECK-DARWIN: "{{.*[\/]}}ld{{(64)?(\.lld)?}}" {{.*}}"-syslibroot" "/path/to/sysroot"
```
Comment (luporl) [inline flang/test/Driver/isysroot.f90:9]: There is a Windows buildbot that uses "ld" only in the linker command line: https://lab.llvm.org/buildbot/#/builders/120/builds/6174.
I guess it's better to remove the `[\/]` part, or make it optional.

Comment (kkwli) [reply flang/test/Driver/isysroot.f90:9]: OK. I will remove `[\/]`. Thanks for catching that.

Comment (kkwli) [inline flang/test/Driver/isysroot.f90:9]:
```suggestion
! CHECK-DARWIN: "{{.*}}ld{{(64)?(\.lld)?}}" {{.*}}"-syslibroot" "/path/to/sysroot"
```

**Pull request merged**